server/resp/src/main/java/org/infinispan/server/resp/RespRequestHandler.java

```diff
@@ -185,15 +185,10 @@ private <E> CompletionStage<RespRequestHandler> stageToReturn(CompletionStage<E>
             "triConsumer was: " + biConsumer + " and handlerWhenComplete was: " + handlerWhenComplete;
       if (CompletionStages.isCompletedSuccessfully(stage)) {
          E result = CompletionStages.join(stage);
-         try {
-            if (handlerWhenComplete != null) {
-               return CompletableFuture.completedFuture(handlerWhenComplete.apply(result));
-            }
-            biConsumer.accept(result, writer);
-         } catch (Throwable t) {
-            writer.error(t);
-            return myStage();
+         if (handlerWhenComplete != null) {
+            return CompletableFuture.completedFuture(handlerWhenComplete.apply(result));
          }
+         biConsumer.accept(result, writer);
          return myStage;
       }
       if (biConsumer != null) {
```

Comment:
I believe we can also remove the try/catch in the `handleAndComposeAsync` lambda below.